historynotificationsmodel.cpp

```diff
@@ -0,0 +1,58 @@
+#include "historynotificationsmodel.h"
+#include <QDebug>
+HistoryNotificationsModel::HistoryNotificationsModel(QObject *parent)
+    : QAbstractListModel{parent} {
+}
+
+int HistoryNotificationsModel::rowCount(const QModelIndex &parent) const {
+    Q_UNUSED(parent);
+    return notificationsList.count();
+}
+
+QVariant HistoryNotificationsModel::data(const QModelIndex &index, int role) const {
+    if(index.row() < 0 || index.row() >= notificationsList.count())
+        return QVariant();
+
+    switch (role) {
+    case Icon:
+        return notificationsList.at(index.row()).img();
+    case Title:
+        return notificationsList.at(index.row()).title();
+    case Message:
+        return notificationsList.at(index.row()).text();
+    case Type:
+        return notificationsList.at(index.row()).type();
+    default:
+        break;
+    }
+    return QVariant();
+}
+
+QHash<int, QByteArray> HistoryNotificationsModel::roleNames() const {
+    QHash<int, QByteArray> roles;
+    roles[Icon]    = "icon";
+    roles[Title]   = "title";
+    roles[Message] = "text";
+    roles[Type]    = "type";
+    return roles;
+}
+
+
+void HistoryNotificationsModel::setHistory(const QmlNotificationService::NotificationData &notificationData) {
+    beginResetModel();
+    notificationsList.push_back(notificationData);
+    endResetModel();
+}
+
+void HistoryNotificationsModel::clearAllHistory() {
+    beginResetModel();
+    notificationsList.clear();
+    endResetModel();
+}
+
+void HistoryNotificationsModel::removeNotificationItemAtIndex(const int elementIndex) {
+    beginResetModel();
```

Comment:
For inserting and removing objects from the list, use the [beginInsertRows](https://doc.qt.io/qt-6/qabstractitemmodel.html#beginInsertRows) and [beginRemoveRows](https://doc.qt.io/qt-6/qabstractitemmodel.html#beginRemoveRows)  methods

See our models on The CheatCard [project](https://quasarapp.ddns.net:3000/QuasarApp/CheatCard/src/branch/next/src/GUI/private/models/cardslistmodel.cpp#L94)